core/drivers/crypto/se050/core/apdu.c

```diff
@@ -16,6 +16,11 @@ TEE_Result crypto_se_do_apdu(enum crypto_apdu_type type,
 {
 	sss_status_t status = kStatus_SSS_Fail;
 
+	if (IS_ENABLED(CFG_CORE_SE05X_I2C_TRAMPOLINE_ONLY) && !se050_session) {
+		/* Defer until REE is ready */
```

Comment:
we should open the session here instead of deferring

Reply:
we should also open in **crypto_se_enable_scp03** (handle the open in that function as well); otherwise se050_enable_scp03 will break looking for the OEFID unless CFG_CORE_SE05X_OEFID is set which is not necessarily true.

Reply:
an to avoid racing while doing it - since we might be coming from a multi-threaded environment - please  use some form of exclusion mechanism

Reply:
try calling se050_session_init() from both places - so no longer a static method - I think that should cover all cases. then when that CFG_CORE_SE05X_I2C_TRAMPOLINE_ONLY is enabled, dont let driver_init call the function